pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -246,6 +247,7 @@ func (c *Cluster) startProvisioningOverPVCs(config *provisionConfig) {
 			placement:        volume.Placement,
 			portable:         volume.Portable,
 			crushDeviceClass: volume.CrushDeviceClass,
+			crushRoot:        opconfig.GetCrushRootFromSpec(&c.spec),
```

Comment:
What may simplify a lot of these changes is if we assume we will pass around the ClusterSpec type instead of the crushRoot string. We already have the cluster spec in most places. This means we don't need to add a `crushRoot` anywhere since `c.spec` is already available to most of the methods. If the cluster spec isn't available to some methods, we can just pass it. Then when the pool needs to extract the crush root, it can call the `GetCrushRootFromSpec()` helper.

Reply:
@travisn Good idea. I did that for the OSD functions.

For the Pool functions (in the client package) I decided against it because they do not yet receive the cluster spec and requiring them to receive that seems odd to me (as they really only need the cluster crush root). If you think differently on this, let me know.